### PR TITLE
Add Step by step example part 1: CIFAR10 Image Stats

### PR DESCRIPTION
**Pull request by chesterxgchen**
### Description

1. Add CIFAR10 image histogram calculation as step-by-step example part 1. 
2. add stats_image job template
3. update job CLI to make sure -debug take effects
4. update job CLI to have special treatment for app_config, app_script keys. as they are only used in client API configurations
5. change the job Config to convert the relative path to absolute path, which fixed downstream bugs
6. change the job CLI to remove the debug job_templates.conf file.
7. add ".conf" in setup.py to make sure the default configs are included. 

A few sentences describing the changes proposed in this pull request.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: examples/hello-world/step-by-step/cifar10/data/download.py [hunk 1]
```diff
@@ -53,6 +53,7 @@ def define_parser():
 
 def main(args):
     datasets.CIFAR10(root=args.dataset_path, train=True, download=True)
+    datasets.CIFAR10(root=args.dataset_path, train=True, download=True)
```
Comment (IsaacYangSLA) [inline examples/hello-world/step-by-step/cifar10/data/download.py:56]: Do we need two same lines here?

**Review by IsaacYangSLA — DISMISSED** [review]
Thanks.  LGTM.

File: examples/hello-world/step-by-step/readme.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+When give an machine learning problem, we probably wonder, 
+where do we start to formulate the federated learning problem. 
+
+* What the data looks like ?
+* How do we analyze the data and compare global statistics with site's data analytics ? 
+* How to formulate the federated algorithms
+  * https://developer.download.nvidia.com/healthcare/clara/docs/federated_traditional_machine_learning_algorithms.pdf
+* Given the formulation, how to convert the existing machine learning or deep learning code to Federated learning code.
+  * [todo: link to ML to FL documentation]( add link to ... )
```
Comment (holgerroth) [inline examples/hello-world/step-by-step/readme.md:11]: misses link

File: examples/hello-world/step-by-step/cifar10/readme.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+
+# Training a image classifier with CIFAR10 data
+ 
+We will use the original [Training a Classifer](https://pytorch.org/tutorials/beginner/blitz/cifar10_tutorial.html) example
+in pytorch as the code base. In the follow examples, we will show various Federated Learning workflows.
```
Comment (holgerroth) [inline examples/hello-world/step-by-step/cifar10/readme.md:5]: "In the follow examples, we ..." should be below the image and description of cifar10.

File: examples/hello-world/step-by-step/cifar10/stats/image_statistics.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        # The feature name is named "red", "gree", "blue" (RGB).
+
+        self.features_ids = {"red": 0, "green": 1, "blue": 2}
+        self.image_features = [
+            Feature("red", DataType.FLOAT),
+            Feature("green", DataType.FLOAT),
+            Feature("blue", DataType.FLOAT),
+        ]
+
+        self.classes = ("plane", "car", "bird", "cat", "deer", "dog", "frog", "horse", "ship", "truck")
```
Comment (holgerroth) [inline examples/hello-world/step-by-step/cifar10/stats/image_statistics.py:52]: Where is this used?

File: examples/hello-world/step-by-step/cifar10/data/download.py [hunk 5]
```diff
@@ -53,6 +53,7 @@ def define_parser():
 
 def main(args):
     datasets.CIFAR10(root=args.dataset_path, train=True, download=True)
+    datasets.CIFAR10(root=args.dataset_path, train=False, download=True)
```
Comment (YuanTingHsieh) [inline examples/hello-world/step-by-step/cifar10/data/download.py:56]: Do we need duplicate lines?
I thought the first line is going to download the dataset?

File: examples/hello-world/step-by-step/readme.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
+#  Step-by-Step Examples
+
+When give an machine learning problem, we probably wonder, 
+where do we start to formulate the federated learning problem.
```
Comment (YuanTingHsieh) [inline examples/hello-world/step-by-step/readme.md:4]: When given a machine learning problem, we probably wonder, 
Where do we start to formulate the federated learning problem?

File: examples/hello-world/step-by-step/readme.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
+#  Step-by-Step Examples
+
+When give an machine learning problem, we probably wonder, 
+where do we start to formulate the federated learning problem. 
+
+* What the data looks like ?
```
Comment (YuanTingHsieh) [inline examples/hello-world/step-by-step/readme.md:6]: What does the data look like?

File: examples/hello-world/step-by-step/readme.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+* How do we analyze the data and compare global statistics with site's data analytics ?
```
Comment (YuanTingHsieh) [inline examples/hello-world/step-by-step/readme.md:7]: How do we analyze the data and compare global statistics with the site's data analytics?

File: job_templates/stats_image/info.md [hunk 9]
```diff
@@ -0,0 +1,11 @@
+# Job Workflow Information Card
+
+## stats_image
+        description = ""Federated image intensity histogram calculation"
+        controller_type = "server"
+        executor_type = "statistics_executor"
+        contributor = "NVIDIA"
+        init_publish_date = "2022"
+        last_updated_date = "2023"
```
Comment (YuanTingHsieh) [inline job_templates/stats_image/info.md:9]: for this job template the publish year is 2023 though

File: job_templates/stats_image/config_fed_server.conf [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+        # statistics result. "stats_writer" is an component defined later in components section.
+        "writer_id": "stats_writer",
+
+        # we are not enable pre_run task.
+        # pre_run is a feature that allows user to run an task before the regular run. Thus called "pre_run".
+        # The pre_run allows use to do some tasks that not necessarily defined by the controller, thus may or may not be
+        # be able to aggregate the global statistics.
+
+        # But this feature enable user to calculate and compute the non-standard metrics and compare different clients
+        # even without globla aggregation.
```
Comment (YuanTingHsieh) [inline job_templates/stats_image/config_fed_server.conf:39]: "global"

File: job_templates/stats_image/config_fed_server.conf [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+  "workflows": [
+    {
+      # id and class path of the StatisticController
+      "id": "fed_stats_controller",
+      "path": "nvflare.app_common.workflows.statistics_controller.StatisticsController",
+      "args": {
+        # the controller specifiy the min_clients and statistic configurations
+        "min_clients": 2,
+        # the statistic_configs defines what are the required local statistics must implement in order for the global
+        # stats to be able to aggregate.
```
Comment (YuanTingHsieh) [inline job_templates/stats_image/config_fed_server.conf:14]: typo: "aggregated"

File: job_templates/sag_pt/info.md [hunk 12]
```diff
@@ -8,5 +8,5 @@
     executor_type = "launcher_executor"
     contributor = "NVIDIA"
     init_publish_date = "2021-01-01" # for unknown date
```
Comment (YuanTingHsieh) [inline job_templates/sag_pt/info.md:10]: "N/A" for unknown date?

File: job_templates/stats_df/info.md [hunk 13]
```diff
@@ -5,7 +5,7 @@
         controller_type = "server"
         executor_type = "statistics_executor"
         contributor = "NVIDIA"
-        init_publish_date = "N/A"
-        last_updated_date = "N/A"
+        init_publish_date = "2023"
+        last_updated_date = "2023"
```
Comment (YuanTingHsieh) [inline job_templates/stats_df/info.md:9]: this is not following the "job_templates/sag_pt/info.md" though? (yyyy-mm-dd)

**Review by SYangster — APPROVED** [review]

**Pull request merged**